src/types/step.ts

```diff
@@ -101,33 +94,19 @@ export interface StepBase {
 
 export interface SwapStep extends StepBase {
   type: 'swap'
-  action: Action
   estimate: Estimate
 }
 
-export function isSwapStep(step: Step): step is SwapStep {
-  return step.type === 'swap'
-}
-
 export interface CrossStep extends StepBase {
   type: 'cross'
-  action: Action
   estimate: Estimate
 }
 
-export function isCrossStep(step: Step): step is CrossStep {
-  return step.type === 'cross'
-}
+export type Step = SwapStep | CrossStep | LifiStep
 
 export interface LifiStep extends StepBase {
   type: 'lifi'
   action: Action
   estimate: Estimate
   includedSteps: Step[]
 }
```

Comment:
The `includedSteps` in `LifiStep` won't be LifiSteps again. So we could remove the recursion here. Try creating
`export type IncludeStep = SwapStep | CrossStep` and add them to LifiStep instead.

Reply:
Thx